rdbms/src/main/java/io/spine/server/storage/jdbc/JdbcStandStorage.java

```diff
@@ -0,0 +1,268 @@
+import io.spine.type.TypeUrl;
+
+import javax.annotation.Nullable;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.Iterables.transform;
+import static com.google.common.collect.Iterators.filter;
+import static com.google.common.collect.Lists.newArrayList;
+import static io.spine.protobuf.TypeConverter.toAny;
+import static io.spine.util.Exceptions.illegalStateWithCauseOf;
+import static java.util.Collections.singleton;
+
+/**
+ * JDBC-based implementation of {@link StandStorage}.
+ *
+ * @author Dmytro Dashenkov
+ */
+public class JdbcStandStorage extends StandStorage {
+
+    private static final EntityQuery<String> EMPTY_QUERY = EntityQueries.from(
+            EntityFilters.getDefaultInstance(),
+            Entity.class
+    );
+
+    private static final Converter<AggregateStateId, String> ID_MAPPER;
+
+    private final JdbcRecordStorage<String> recordStorage;
+
+    static {
+        try {
+            // Ensure the AggregateStateId Stringifier is registered.
+            Class.forName(AggregateStateId.class.getCanonicalName());
+        } catch (ClassNotFoundException e) {
+            throw illegalStateWithCauseOf(e);
+        }
+        final Optional<Stringifier<AggregateStateId>> stringifier =
+                StringifierRegistry.getInstance().get(AggregateStateId.class);
+        checkState(stringifier.isPresent(), "AggregateStateId Stringifier is not registered!");
+        ID_MAPPER = stringifier.get();
+    }
+
+    protected JdbcStandStorage(Builder builder) {
+        super(builder.isMultitenant());
+        recordStorage = JdbcRecordStorage.<String>newBuilder()
+                                         .setDataSource(builder.getDataSource())
+                                         .setMultitenant(builder.isMultitenant())
+                                         .setEntityClass(StandEntity.class)
+                                         .build();
+    }
+
+    @Override
+    public Iterator<AggregateStateId> index() {
+        final Iterator<String> index = recordStorage.index();
+        final Iterator<AggregateStateId> result = Iterators.transform(index, ID_MAPPER.reverse());
+        return result;
+    }
+
+    @Override
+    public Iterator<EntityRecord> readAllByType(TypeUrl type) {
+        return readAllByType(type, FieldMask.getDefaultInstance());
+    }
+
+    @Override
+    public Iterator<EntityRecord> readAllByType(final TypeUrl type, FieldMask fieldMask) {
+        final Iterator<EntityRecord> allRecords = readAll(fieldMask);
+        final String requiredTypeUrl = type.value();
+        final Iterator<EntityRecord> result = filter(allRecords, new Predicate<EntityRecord>() {
+            // TODO:2017-07-14:dmytro.dashenkov: Replace in-memory filtering with SQL query.
```

Comment:
Please have issues (and links should go here) for every `TODO`, that still resides.